### PR TITLE
HDDS-7815. Extract S3 secret layer

### PR DESCRIPTION
**Pull request by Pochatkin**
## What changes were proposed in this pull request?

Extract interfaces of S3 secret management. This changes required to HDDS-7814. 
No new functionality introducing, only code refactoring and encapsulation. 

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7815

## How was this patch tested?
Existed test passed.

### CODE REVIEW

**Comment by kerneltime** [comment]
Can you look into the CI failures? 
https://github.com/apache/ozone/actions/runs/3968184281/jobs/6801029715 (and others)
```
Error:  /home/runner/work/ozone/ozone/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/security/S3RevokeSecretRequest.java:[111,24] method invalidateCacheEntry in interface org.apache.hadoop.ozone.om.S3SecretManager cannot be applied to given types;
[1706](https://github.com/apache/ozone/actions/runs/3968184281/jobs/6801029715#step:6:1707)
  required: java.lang.String,long
```

**Comment by adoroszlai** [comment]
@Pochatkin Before opening a PR, please wait for successful CI run in your fork. This helps save resources for Apache project. In this case checkstyle is failing and needs a fix.  Please update the `master` branch in your fork from `apache/ozone`, then enable the [build-branch](https://github.com/Pochatkin/ozone/actions/workflows/post-commit.yml) workflow in your fork.

**Comment by Pochatkin** [comment]
How I can restart failed test? Looks like this is flaky

**Comment by neils-dev** [comment]
> How I can restart failed test? Looks like this is flaky

In the PR goto the Checks view.  Within the Checks select the "build-branch", after click on the "re-run jobs" button in the upper right corner of the view.

**Comment by Pochatkin** [comment]
@neils-dev @ChenSammi Could you please review? 'm trying to figure out test fails but it all looks like flakes.

**Comment by myskov** [comment]
@ChenSammi @neils-dev please take a look at PR

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/S3SecretStoreMap.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  @Override
+  public void revokeSecret(String kerberosId) throws IOException {
+    map.remove(kerberosId);
+  }
+
+  @Override
+  public S3Batcher batcher() {
+    return null;
+  }
+}
```
Comment (ChenSammi) [inline hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/S3SecretStoreMap.java:55]: Please move this implementation to test package if it only for test.

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/S3SecretStore.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ */
+public interface S3SecretStore {
+
+  /**
+   * Store provided s3 secret with associated kerberos ID.
+   * @param kerberosId kerberos ID.
+   * @param secret s3 secret.
+   * @throws IOException in case when secret storing failed.
+   */
+  void store(String kerberosId, S3SecretValue secret) throws IOException;
```
Comment (ChenSammi) [inline hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/S3SecretStore.java:34]: Consistent naming is preferred,  either (store, get, revoke) or (storeSecret, getSecret, revokeSecret).  Consider OmMetadataManagerImpl implements the S3SecretStore. And there are many functions in OmMetadataManagerImpl. I more prefer (storeSecret, getSecret, revokeSecret) .

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/S3SecretManagerImpl.java [hunk 3]
```diff
@@ -88,22 +64,43 @@ public String getS3UserSecretString(String awsAccessKey)
...
   }
 
-  public OMMetadataManager getOmMetadataManager() {
-    return omMetadataManager;
+  @Override
+  public void store(String kerberosId, S3SecretValue secretValue)
+      throws IOException {
+    s3SecretStore.store(kerberosId, secretValue);
+    if (LOG.isTraceEnabled()) {
+      LOG.trace("Secret for accessKey:{}, proto:{}", kerberosId, secretValue);
```
Comment (ChenSammi) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/S3SecretManagerImpl.java:81]: secretValue should not be logged.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/security/S3GetSecretRequest.java [hunk 4]
```diff
@@ -154,24 +147,20 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       assert (accessId.equals(updateGetS3SecretRequest.getKerberosID()));
     }
 
+    S3SecretManager s3SecretManager = ozoneManager.getS3SecretManager();
     try {
-      acquiredLock = omMetadataManager.getLock()
-          .acquireWriteLock(S3_SECRET_LOCK, accessId);
-
```
Comment (ChenSammi) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/security/S3GetSecretRequest.java:160]: This lock protects both getS3Secret and cache operation. It cannot be removed.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/s3/security/OMSetSecretResponse.java [hunk 5]
```diff
@@ -1,25 +1,26 @@
...
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+  <p>
+  http://www.apache.org/licenses/LICENSE-2.0
+  <p>
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
```
Comment (ChenSammi) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/s3/security/OMSetSecretResponse.java:16]: Please revert this piece of change.

**Review by ChenSammi — COMMENTED** [review]
Thanks @Pochatkin for abstract the secret storage layer. The overall patch looks good.

**Comment by ChenSammi** [comment]
The last patch LGTM +1.  Thanks @Pochatkin for the contribution.

**Pull request merged**